contracts/token/ERC20/extensions/IERC20Permit.sol

```diff
@@ -10,6 +10,26 @@ pragma solidity ^0.8.20;
  * Adds the {permit} method, which can be used to change an account's ERC20 allowance (see {IERC20-allowance}) by
  * presenting a message signed by the account. By not relying on {IERC20-approve}, the token holder account doesn't
  * need to send a transaction, and thus is not required to hold Ether at all.
+ *
+ * ==== Security Considerations
+ *
+ * There are two important considerations concerning the use of `permit`. The first is that a valid permit signature
+ * expresses an allowance, and it should not be assumed to convey additional meaning. In particular, it should not be
+ * considered as an intention to spend the approval in any specific way. The second is that because permits have
+ * built-in replay protection and can be submitted by anyone, they can be frontrun. A protocol that uses permits should
+ * take this into consideration and allow a `permit` call to fail. Combining these two aspects, a good pattern that can
+ * be generally recommended is:
```

Comment:
We should not recommend the pattern without explciitelly saying what we recommend it for !!!

People see its recommended, use it, and then don't understand that it doesn't do what they expect it to do. IMO we should say: "if your usecase is X, and you understand that Y, then we recommend using the following pattern"

Reply:
Do you say that because it says "_generally_ recommended"? Can you think of any scenarios where this would not be recommended?

I think the recommendation should apply always, "unless you know what you're doing".

An example of something different might be a protocol like CowSwap, where a solver may submits your permit along with the trade, so it will not be `msg.sender`. But this is exactly the kind of thing where you need to be very careful about how you design it so I put it in the "you better know what you're doing" bin.

Reply:
> Can you think of any scenarios where this would not be recommended?

It really demands what you are trying to achieve as a dev. For example if you try to do a "one step deposit" that set the approval through permit, then consume it ... you may be tempted to do a try catch on the permit, than consume all the approval you have (if any). I think that is a bad design, that your recommandation does address. People might think its a good idea because you recommand something that allows that.

Reply:
>that your recommandation does address

Do you mean "does not address"?

Does this example shown in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4582/files#r1319293034 helpe?

Reply:
I did mean "does address".

Reply:
I think that the current wording is ok.